lib/schema_tools/modules/object_properties.rb

```diff
@@ -0,0 +1,32 @@
+module SchemaTools
+  module Modules
+    module ObjectProperties
+      PROPERTIES = 'properties'
+      ALLOF = 'allOf'
+      ANYOF = 'anyOf'
+      ONEOF = 'oneOf'
+      PROPERTY_CONTAINERS = [ PROPERTIES, ALLOF, ANYOF, ONEOF ]
+
+      def all_properties(schema_hash, properties_hash = ActiveSupport::HashWithIndifferentAccess.new)
```

Comment:
Could you clarify why we need to pass the `properties_hash` param? I'm not seeing it passed when you call `all_properties` in this PR.